zeppelin-zengine/pom.xml

```diff
@@ -166,7 +249,14 @@
     <dependency>
       <groupId>com.google.guava</groupId>
       <artifactId>guava</artifactId>
-      <version>15.0</version>
+      <!-- Android is a bit of a misnomer, it is for Java 7 -->
+      <version>23.0-android</version>
```

Comment:
Any reason to upgrade guava ? Guava's compatibility across versions is very weak, it is very likely to cause side effect when upgrading guava.

Reply:
Unfortunately, yes. `google-cloud-java` seems to depend on things in 23.0. When I tried excluding guava from google-cloud-java, I got method not found errors at runtime.

That being said, if all of zeppelin's tests pass without hitting class not found / method not found errors, is it okay to upgrade Guava?

Reply:
Apparently, many travis CI fails.

Reply:
There's just one issue (causing two tests to fail):

```
java.lang.NoSuchMethodError: com.google.common.base.Objects.toStringHelper(Ljava/lang/Object;)Lcom/google/common/base/Objects$ToStringHelper;
	at org.apache.hadoop.metrics2.lib.MetricsRegistry.toString(MetricsRegistry.java:406)
	at java.lang.String.valueOf(String.java:2994)
	at java.lang.StringBuilder.append(StringBuilder.java:131)
	at org.apache.hadoop.ipc.metrics.RpcMetrics.<init>(RpcMetrics.java:74)
	at org.apache.hadoop.ipc.metrics.RpcMetrics.create(RpcMetrics.java:80)
	at org.apache.hadoop.ipc.Server.<init>(Server.java:2218)
	at org.apache.hadoop.ipc.RPC$Server.<init>(RPC.java:951)
	at org.apache.hadoop.ipc.ProtobufRpcEngine$Server.<init>(ProtobufRpcEngine.java:534)
	at org.apache.hadoop.ipc.ProtobufRpcEngine.getServer(ProtobufRpcEngine.java:509)
	at org.apache.hadoop.ipc.RPC$Builder.build(RPC.java:796)
	at org.apache.hadoop.hdfs.server.namenode.NameNodeRpcServer.<init>(NameNodeRpcServer.java:351)
	at org.apache.hadoop.hdfs.server.namenode.NameNode.createRpcServer(NameNode.java:674)
	at org.apache.hadoop.hdfs.server.namenode.NameNode.initialize(NameNode.java:647)
	at org.apache.hadoop.hdfs.server.namenode.NameNode.<init>(NameNode.java:812)
	at org.apache.hadoop.hdfs.server.namenode.NameNode.<init>(NameNode.java:796)
	at org.apache.hadoop.hdfs.server.namenode.NameNode.createNameNode(NameNode.java:1493)
	at org.apache.hadoop.hdfs.MiniDFSCluster.createNameNode(MiniDFSCluster.java:1115)
	at org.apache.hadoop.hdfs.MiniDFSCluster.createNameNodesAndSetConf(MiniDFSCluster.java:986)
	at org.apache.hadoop.hdfs.MiniDFSCluster.initMiniDFSCluster(MiniDFSCluster.java:815)
	at org.apache.hadoop.hdfs.MiniDFSCluster.<init>(MiniDFSCluster.java:475)
	at org.apache.hadoop.hdfs.MiniDFSCluster$Builder.build(MiniDFSCluster.java:434)
	at org.apache.zeppelin.interpreter.MiniHadoopCluster.start(MiniHadoopCluster.java:41)
	at org.apache.zeppelin.interpreter.SparkInterpreterModeTest.setUp(SparkInterpreterModeTest.java:29)
```

This is very unfortunate. Hadoop's `trunk` and `branch-2` no longer use this (now deprecated) class: `https://github.com/apache/hadoop/blob/branch-2/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/lib/MetricsRegistry.java`. Zeppelin depends on Hadoop 2.7.

Reply:
The fix is in Hadoop 2.9.0 and Hadoop 2.8.3 https://issues.apache.org/jira/browse/HADOOP-14891